OpenRA.Game/Actor.cs

```diff
@@ -41,12 +41,23 @@ public sealed class Actor : IScriptBindable, IScriptNotifyBind, ILuaTableBinding
 		public int Generation;
 
 		Lazy<Rectangle> bounds;
+		Lazy<Rectangle> visualBounds;
 		Lazy<IFacing> facing;
 		Lazy<Health> health;
 		Lazy<IOccupySpace> occupySpace;
 		Lazy<IEffectiveOwner> effectiveOwner;
 
 		public Rectangle Bounds { get { return bounds.Value; } }
+
+		public Rectangle VisualBounds
+		{
+			get
+			{
+				var sd = Info.Traits.GetOrDefault<ISelectionDecorationsInfo>();
+				return (sd != null && sd.SelectionBoxBounds != null) ? visualBounds.Value : bounds.Value;
```

Comment:
Sorry, this will look of traits each time you select an actor, for each actor.
What I meant was check in `visualBounds` ([here](https://github.com/OpenRA/OpenRA/pull/8388/files#diff-25e3f19d154f93bcb747e9f12c252028R126)) and if there is no such trait, give it `bounds`' value (or make it `null` and then do `if (visualBounds == null) return bounds.Value;` here).